src/core/ddsc/include/dds/dds.h

```diff
@@ -3988,6 +4265,115 @@ dds_take_with_collector (
  */
 #define DDS_HAS_READCDR 1
 
+/**
+ * @brief Access the collection of serialized data values (of same type) and
+ *        sample info from the data reader, readcondition or querycondition
+ *        without updating state.
+ * @ingroup reading
+ * @component read_data
+ *
+ * This call accesses the serialized data from the data reader, readcondition or
+ * querycondition and makes it available to the application. The serialized data
+ * is made available through @ref ddsi_serdata structures. Returned samples are
+ * marked not marked as READ.
+ *
+ * When using a readcondition or querycondition, their masks are or'd with the given mask.
+ *
+ * If the sample/view/instance state component in the mask is 0 and there is no read or query condition,
+ * to combine it with, it is treated as equivalent to any sample/view/instance state.
+ *
+ * Return value provides information about the number of samples read, which will
+ * be <= maxs. Based on the count, the buffer will contain serialized data to be
+ * read only when valid_data bit in sample info structure is set.
+ * The buffer required for data values, could be allocated explicitly or can
+ * use the memory from data reader to prevent copy. In the latter case, buffer and
+ * sample_info should be returned back, once it is no longer using the data.
```

Comment:
Not a problem initiated by this PR, and the paragraph has been used in `dds.h` for a while for several read functions, but I think it's not completely clear under what conditions the caller should return the buffer (and not sample_info..). And shouldn't the `buf` parameter be `in,out`?

Reply:
Good point, I can't even make sense of that description myself! There'll be a separate PR for the documentation of those functions shortly.